ACE/tests/Message_Queue_Test_Ex.cpp

```diff
@@ -709,6 +709,98 @@ int queue_priority_test (ACE_Message_Queue_Ex<User_Class, ACE_SYNCH>& q)
   return status;
 }
 
+class Queue_Ex_Iterator_No_Lock
+ : public ACE_Message_Queue_Iterator<ACE_SYNCH, ACE_System_Time_Policy>
+{
+  typedef ACE_Message_Queue_Iterator<ACE_SYNCH, ACE_System_Time_Policy> inherited;
+
+ public:
+  typedef ACE_Message_Queue_Ex<User_Class, ACE_SYNCH, ACE_System_Time_Policy> MESSAGE_QUEUE_EX_T;
+
+  explicit Queue_Ex_Iterator_No_Lock (MESSAGE_QUEUE_EX_T& queue_in)
+   : inherited (queue_in.queue ())
+  {}
+  virtual ~Queue_Ex_Iterator_No_Lock () {}
+
+  int next (User_Class*& message_inout)
+  {
+    if (inherited::curr_)
+    {
+      message_inout = reinterpret_cast<User_Class*> (inherited::curr_->base ());
+      return 1;
+    } // end IF
+
+    return 0;
+  }
+  int done (void) const { return (!inherited::curr_ ? 1 : 0); }
+  int advance (void)
```

Comment:
Add empty line between the operations. Return `bool` for done? Remove redundant `(void)` for no arguments, just `()`

Reply:
ok